packages/amplify-e2e-core/src/utils/rds.ts

```diff
@@ -241,6 +244,83 @@ export const createRDSCluster = async (config: RDSConfig): Promise<ClusterInfo>
   }
 };
 
+/**
+ * Setup the test database and data in the pre-existing RDS Aurora serverless V2 cluster with one writer DB instance. Get the necessary configuration settings of the cluster and instance.
+ * @param identifier Cluster idenfitier.
+ * @param config Configuration of the database cluster.
+ * @param queries Initial queries to be executed.
+ * @returns Cluster configuration information.
+ */
+export const setupDataInExistingCluster = async (identifier: string, config: RDSConfig, queries: string[]): Promise<ClusterInfo> => {
+  try {
+    const client = new RDSClient({ region: config.region });
+    const describeClusterResponse = await client.send(
+      new DescribeDBClustersCommand({ Filters: [{ Name: 'db-cluster-id', Values: [identifier] }] }),
+    );
+    if (!describeClusterResponse || !describeClusterResponse?.DBClusters[0]) {
+      throw Error('Specified cluster info cannot be fetched');
+    }
+    const dbClusterObj = describeClusterResponse.DBClusters[0];
+    const instances = dbClusterObj?.DBClusterMembers;
+    if (!instances || instances?.length === 0) {
+      throw new Error('No instances are present in the specified cluster');
+    }
+    const instanceId = instances[0]?.DBInstanceIdentifier;
+    const describeInstanceCommand = new DescribeDBInstancesCommand({ DBInstanceIdentifier: instanceId });
+    const describeInstanceResponse = await client.send(describeInstanceCommand);
+    if (!describeInstanceResponse || describeInstanceResponse?.DBInstances?.length === 0) {
+      throw Error('Specified cluster instance info cannot be fetched');
+    }
+
+    const clusterArn = dbClusterObj.DBClusterArn;
+    const secretArn = dbClusterObj.MasterUserSecret.SecretArn;
+    const defaultDbName = dbClusterObj.DatabaseName;
+    const dataClient = new RDSDataClient({ region: config.region });
+    const createDBInput: ExecuteStatementCommandInput = {
```

Comment:
As mentioned in previous PRs: we need to sanitize the db name before interpolating it into the SQL. I believe you've currently got the sanitization going on in the test code, but it more appropriately belongs here, especially since you're returning the dbName as part of your return value anyway.